File: app/src/main/java/line/javafoundation/producer_consumer/Producer.java
package line.javafoundation.producer_consumer;

public class Producer implements Runnable {
    private Storage<Phone> storage;

    public Producer(Storage<Phone> storage) {
        this.storage = storage;
    }

    public void run() {
        for (int i = 0; i < 20; i++) {
            storage.produce(new Phone(i));
            try {
                Thread.sleep(100);
//                每隔1000毫秒生产一个产品
            } catch (InterruptedException e) {
//                TODO Auto -generated catch block e.printStackTrace();
            }
        }
    }
}

File: javalib/src/main/java/com/yin/javalib/producer_consumer/Producer.kt
package com.yin.javalib.producer_consumer

class Producer(private val storage: Storage<Phone>) : Runnable {
    override fun run() {
        for (i in 0..19) {
            storage.produce(Phone(i))
            try {
                Thread.sleep(1000)
                //                每隔1000毫秒生产一个产品
            } catch (e: InterruptedException) {
            }
        }
    }

}